### PR TITLE
feat(eyeglass): Adds new package that enables simple Eyeglass support.

### LINKED ISSUE #398: Productize Eyeglass support
Right now we're doing this in `css-blocks.config.js` to add eyeglass support to a project:

```js
const sass = require("node-sass");
const eyeglass = require("eyeglass");
function scss(file, data, _configuration, _sourceMap) {
  return new Promise((resolve, reject) => {
    const sassOptions = {
      file,
      data,
      outputStyle: "expanded",
      sourceMap: true,
      outFile: file.replace(/scss$/, "css"),
    };
    sass.render(eyeglass(sassOptions), (err, res) => {
      if (err) {
        reject(err)
      } else {
        resolve({
          content: res.css.toString(),
          sourceMap: res.map.toString(),
          dependencies: res.includedFiles,
        })
      }
    });
  })
}
module.exports = {
  preprocessors: { scss }
};
```

Instead we should be able to do this:

```js
const sass = require("node-sass");
const eyeglass = require("eyeglass");
const eyeglassAdaptor = require("@css-blocks/eyeglass");

const sassOptions = {
  outputStyle: "expanded",
};

const scss = eyeglassAdaptor(sass, eyeglass, sassOptions);

module.exports = {
  preprocessors: { scss }
};
```

A few things to note here: Sass and Eyeglass are dependency injected. I don't want us to manage what versions or implementations get installed.

The options other than `outputStyle` that we're setting need to be set that way, we shouldn't allow them to be overridden by the caller so we should `Object.assign` into their options, not the other way around.

### PR DESCRIPTION
**Pull request by abritinthebay**
Fixes #398 

Should be a cleaner TS implementation than my previous PR (thanks for the feedback and help using Namespaces as types). I'm using Sinon spies in the tests, and because of that found I didn't need to use the module importer mock you mentioned (as rather than testing all of css-blocks in the unit tests I'm just testing what _this package_  takes in and puts out, as well as that the created functions are called). 

Code test coverage is 100%, which is nice.

### CODE REVIEW

File: packages/@css-blocks/eyeglass/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { ProcessedFile } from "@css-blocks/core";
+import Eyeglass from "eyeglass"; // works, even tho a cjs export. huh.
+import { Options, Result, SassError } from "node-sass";
+import SassImplementation from "node-sass";
+
+export function adaptor(foo: typeof SassImplementation, eyeglass: typeof Eyeglass, options: Options = {}) {
```
Comment (chriseppstein) [inline packages/@css-blocks/eyeglass/src/index.ts]: let's call it `sass` instead of `foo`.

File: packages/@css-blocks/eyeglass/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@css-blocks/eyeglass",
+  "author": "Gregory Wild-Smith <gwildsmith@linkedin.com>",
+  "description": "Tool to provide easy interoperation with the Eyeglass SASS module manager",
+  "license": "BSD-2-Clause",
+  "version": "1.0.0-alpha.6",
```
Comment (chriseppstein) [inline packages/@css-blocks/eyeglass/package.json]: we're at alpha.7 now.

File: packages/@css-blocks/eyeglass/src/index.ts (hunk 1 above)
Comment (abritinthebay) [reply packages/@css-blocks/eyeglass/src/index.ts]: ... argh. Yes 🤦‍♂

**Comment by abritinthebay** [comment]
Updated with those couple of small changes 😄

**Pull request merged**